### PR TITLE
Add 2 more tag validation checks

### PR DESCRIPTION
**Pull request by nckturner**
* ensure we are on a release branch
* ensure the major/minor versions match the release branch

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

### CODE REVIEW

File: hack/tag-release.sh [hunk 1]
```diff
@@ -21,6 +21,21 @@ if [[ ! "${VERSION}" =~ ^([0-9]+[.][0-9]+)[.]([0-9]+)(-(alpha|beta)[.]([0-9]+))?
   exit 1
 fi
 
+BRANCH=$(git branch --show-current)
```
Comment (nnmin-aws) [inline hack/tag-release.sh:24]: LGTM!
thank you

File: hack/tag-release.sh [hunk 2]
```diff
@@ -21,6 +21,21 @@ if [[ ! "${VERSION}" =~ ^([0-9]+[.][0-9]+)[.]([0-9]+)(-(alpha|beta)[.]([0-9]+))?
...
+
+if [[ ! "${BRANCH}" =~ ^(release-)([0-9]+[.][0-9]+)$ ]]; then
+    echo "Automatic tag creation must take place on a release branch."
+    exit 1
+fi
+
+BRANCH_MAJ_MIN=${BRANCH#release-}
+VERSION_MAJ_MIN=$(echo ${VERSION} | sed -Ee 's/-(alpha|beta)\.[0-9]+$//' | sed -Ee 's/\.[0-9]+$//')
```
Comment (nckturner) [inline hack/tag-release.sh:32]: First remove something like `-alpha.0` if it exists, second remove the patch version.

Comment (wongma7) [reply hack/tag-release.sh:32]: could be easier/clearer to just grep the "vX.Y" tbh ? but lgtm anyway

**Pull request merged**